### PR TITLE
feat(generator): Add tailwind in generator MANAGER-11981

### PR DESCRIPTION
**Pull request by stif59100**
| Question         | Answer
| ---------------- | ---
| Branch?          | `dev-transformation
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11981
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Related

I added class tailwind on generator

### CODE REVIEW

File: packages/manager/core/generator/app/templates/src/global.css.hbs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@tailwind base;
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/core/generator/app/templates/src/global.css.hbs:1]: why .hbs extension file ? we dont need to replace variable no ?

File: packages/manager/core/generator/app/templates/src/components/Error/Error.tsx [hunk 2]
```diff
@@ -39,7 +38,7 @@ function getTrackingTypology(error: errorMessage) {
 }
 
 const ErrorBanner: React.FC<ErrorObject> = ({ error }) => {
-  const { t } = useTranslation('{{appName}}/error');
+  const { t } = useTranslation('aaaa/error');
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/Error/Error.tsx:41]: Put the variable back

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/TileCustom.tsx [hunk 3]
```diff
@@ -14,8 +13,8 @@ const TileCustom: React.FC<React.PropsWithChildren<SectionProps>> = ({
   title,
   children,
 }) => (
-  <OsdsTile flex>
-    <div className="tile-content">
+  <OsdsTile className="border-2 border-solid border-gray-200 w-full" flex>
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/TileCustom.tsx:16]: why adding border color ?

File: packages/manager/core/generator/app/templates/src/components/Error/Error.tsx (hunk 2 above)
Comment (aboungnaseng-ovhcloud) [inline packages/manager/core/generator/app/templates/src/components/Error/Error.tsx:41]: reverse this please

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx [hunk 4]
```diff
@@ -36,7 +38,7 @@ const Datagrid: React.FC<Props> = ({ serviceKey, data }) => {
               </OsdsText>
             </th>
           ))}
-          <th>
+          <th className="p-5 bg-cyan-200">
```
Comment (Kranysys) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx:41]: We need can maybe use ODS tokens for color
https://ovh.github.io/design-system/latest/?path=/story/code-design-tokens-blue-jeans-colors--page
Like `--ods-blue-jeans-color-primary-300`

Comment (chipp972) [reply packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx:41]: Agreed. Except the variable name is `--ods-color-primary-300`

File: packages/manager/core/generator/app/templates/src/components/ButtonTooltip/ButtonTooltip.tsx [hunk 5]
```diff
@@ -62,10 +61,12 @@ const ButtonTooltip: React.FC<ButtonTooltipProps> = (props) => {
         />
       </OsdsButton>
       {showTooltip && (
-        <div ref={tooltipRef} className="tooltip">
-          <div className="tooltip-text">
+        <div ref={tooltipRef} className="relative inline-block left-[-80%]">
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/ButtonTooltip/ButtonTooltip.tsx:64]: Can't we use the tooltip from ODS ?

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 6]
```diff
@@ -22,23 +20,25 @@ export type DashboardLayoutProps = {
 };
 
 const Dashboard: React.FC<DashboardLayoutProps> = ({ tabs }) => {
-  const [panel, setActivePanel] = useState('');
+  const [activeTab, setActiveTab] = useState<string>('');
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx:23]: shouldn't we initialize to the first tab ?

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 7]
```diff
@@ -22,23 +20,25 @@ export type DashboardLayoutProps = {
...
+    );
+    if (activeTabFromLocation) {
+      setActiveTab(activeTabFromLocation.to);
     } else {
-      setActivePanel(tabs[0].name);
+      setActiveTab('');
       navigate(`${tabs[0].to}`);
     }
-  }, [location.pathname]);
+  }, [location.pathname, tabs, navigate]);
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx:37]: if you put tabs and navigate, it's like you didn't use a `useEffect` here

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx (hunk 6 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx:23]: We should'nt initialize cause we didn't want /Tabs1 in the url, by default it display index.tsx file

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx (hunk 7 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx:37]: sure can delete it please ?

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx (hunk 4 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx:41]: Yes, in the next weeks, we will delete this to replace by the ods component

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx [hunk 8]
```diff
@@ -36,7 +39,7 @@ const Datagrid: React.FC<Props> = ({ serviceKey, data }) => {
               </OsdsText>
             </th>
           ))}
-          <th>
+          <th className="p-5 background-ods-primary-100">
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.tsx:42]: background-ods-primary-100 is working ?

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.scss [hunk 9]
```diff
@@ -1,15 +1,11 @@
...
+  background-color: var(--ods-color-primary-100);
+}
 
-  tr {
-    background-color: #fff;
-    border: 1px solid #bef1ff;
-    border-collapse: collapse;
-  }
+.background-ods-primary-000 {
+  background-color: var(--ods-color-primary-000);
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/dataGrid.scss:10]: Don't create utility classes if the file is specific to dataGrid.
If you create utility classes, they should be global.

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Review by Kranysys — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

**Pull request merged**